plugins/keyboard-navigation/test/navigation_modify_test.mocha.js

```diff
@@ -0,0 +1,476 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+const chai = require('chai');
+const Blockly = require('blockly/node');
+const {Navigation} = require('../src/navigation');
+const assert = chai.assert;
+
+suite('Insert/Modify', function() {
+  /**
+   * Captures the strings sent to console.warn() when calling a function.
+   * Copies from core.
+   * @param {function} innerFunc The function where warnings may called.
+   * @return {string[]} The warning messages (only the first arguments).
+   */
+  function captureWarnings(innerFunc) {
+    const msgs = [];
+    const nativeConsoleWarn = console.warn;
+    try {
+      console.warn = function(msg) {
+        msgs.push(msg);
+      };
+      innerFunc();
+    } finally {
+      console.warn = nativeConsoleWarn;
+    }
+    return msgs;
+  }
+
+  /**
+   * Check that modify failed.
+   * @param {Navigation} navigation The class under test.
+   * @param {Blockly.WorkspaceSvg} workspace The main workspace.
+   * @param {!Blockly.ASTNode} markerNode The node to try to connect to.
+   * @param {!Blockly.ASTNode} cursorNode The node to connect to the markerNode.
+   */
+  function assertModifyFails(navigation, workspace, markerNode, cursorNode) {
+    let modifyResult;
+    const warnings = captureWarnings(function() {
+      modifyResult = navigation.modify(workspace, markerNode, cursorNode);
+    });
+    assert.isFalse(modifyResult);
+    assert.equal(
+        warnings.length, 1, 'Expecting 1 warnings for why modify failed.');
```

Comment:
I would recommend also adding an assertion checking the warning text.
You could use `chai.assert.match` to check part of the message, so we can reasonably conclude that it's the message we were expecting.

Reply:
So I did go through and start to update this, but I think because of how experimental it is I'm not sure I want to add this check. There wasn't a lot of reasoning behind the wording we chose for the error messages and I expect us to change them in the future. I think it would be more helpful to add them back in once we settle on the wording we want.